### PR TITLE
Switch to Parcel for building and adjust export path and build

### PR DESCRIPTION
**Pull request by holblin**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jogibear9988** [comment]
do you still add file extensions to imports in the esm build?

**Comment by holblin** [comment]
Hi @jogibear9988 , I do not understand your question. Could you rephrase it and give me an example with the current result and the expected result if there is a concern or a problem?

**Comment by holblin** [comment]
@jogibear9988 , I will address potential feedback on the next `beta` / `rc` version

**Comment by jogibear9988** [comment]
For me this now does not work, cause I want to use it in a VSCode extension, wich does not allow .mjs at the moment.
But I created an issue for this a vscode:
https://github.com/microsoft/vscode/issues/176567

**Comment by jogibear9988** [comment]
But I think the mime type for .mjs/.cjs could more often be a problem. I think to move it to two folders with .js extension would be more error prune

**Comment by holblin** [comment]
Would that be a problem for you to use version `4.2.x` until https://github.com/microsoft/vscode/pull/176576 is merged?
I could also, do the two folder things if the vscode PR doesn't get some traction (but that would be better if they fix it on their side).

**Comment by jogibear9988** [comment]
for me it's now not a problem any more, but I wanted to show that it could be a problem for more people, cause mime type support dor these extensions is often not so good
